go-packages/meep-logger/logger.go

```diff
@@ -17,59 +17,117 @@
 package logger
 
 import (
-	log "github.com/sirupsen/logrus"
+	"net/http"
+	"time"
+	"runtime"
+	"path"
+	"fmt"
+	logrus "github.com/sirupsen/logrus"
 )
 
 var componentName string
 
 type Fields map[string]interface{}
 
 func MeepTextLogInit(name string) {
-	log.SetFormatter(&log.TextFormatter{})
-	log.SetLevel(log.DebugLevel)
+	//logrus.SetLevel(logrus.TraceLevel)
+        logrus.SetLevel(logrus.DebugLevel)
 	componentName = name
 }
 
 func MeepJSONLogInit(name string) {
-	log.SetFormatter(&log.JSONFormatter{})
-	log.SetLevel(log.DebugLevel)
+	logrus.SetFormatter(&logrus.JSONFormatter{})
+	//logrus.SetLevel(logrus.TraceLevel)
+        logrus.SetLevel(logrus.DebugLevel)
 	componentName = name
 }
 
+// getLogCaller
+// setReportCaller cannot ignore caller levels, so cannot work in a wrapper. Fix not merged in github, so doing it manually
+func getLogCaller() string {
+	_, file, line, _ := runtime.Caller(2)
+
+        location := fmt.Sprintf("%v:%v", path.Base(file), line)
+	return location
+}
+
 func Info(args ...interface{}) {
-	log.WithFields(log.Fields{
+	logrus.WithFields(logrus.Fields{
 		"meep.component": componentName,
+		"meep.time": time.Now().String(),
```

Comment:
Is this timestamp in addition to the default timestamp from logrus? If so, we should not have 2 timestamps.

Also, can't you use the logrus TimestampFormat & formatters to set a more precise full timestamp?

Reply:
the reason for both timestamp was actually the fact that we wanted more precision, I will check

Reply:
using the formatter.. but its either seconds or ns... i then set it to ns precision, only one time required

Reply:
Formatter seems to do the job. Also it looks like you can set microsecond granularity using six 0 or 9 after the point.

Reply:
true... it is possible to use finer granularity... was changed to ms, although we could have had us or ns... can always be changed in the future if more granularity is required, but ms seems appropriate.